processing/src/main/java/org/apache/carbondata/processing/newflow/sort/impl/UnsafeBatchParallelReadMergeSorterImpl.java

```diff
@@ -197,7 +209,12 @@ private void createSortDataRows() {
 
     @Override public UnsafeSingleThreadFinalSortFilesMerger next() {
       try {
-        return mergerQueue.take();
+        UnsafeSingleThreadFinalSortFilesMerger unsafeSingleThreadFinalSortFilesMerger =
```

Comment:
@ravipesala  Please review portion 
Here i have fixed the dead lock issue
https://issues.apache.org/jira/browse/CARBONDATA-913